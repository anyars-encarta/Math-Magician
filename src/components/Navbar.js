import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from '../modules/Navbar.module.css';

function Navbar() {
  return (
    <div className={styles.navbar}>
      <h1 className={styles.header}>Encarta Math Magicians</h1>
      <nav>
        <li><NavLink className={styles.navLink} to="/">Home</NavLink></li>
        <li><NavLink className={styles.navLink} to="/calculator">Calculator</NavLink></li>
        <li><NavLink className={styles.navLink} to="/quote">Quote</NavLink></li>
      </nav>
    </div>
  );
}

export default Navbar;
