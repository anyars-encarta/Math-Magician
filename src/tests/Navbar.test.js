import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('Navbar', () => {
  test('renders the Navbar component correctly', () => {
    render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(screen.getByText('Encarta Math Magicians')).toBeInTheDocument();
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Calculator')).toBeInTheDocument();
    expect(screen.getByText('Quote')).toBeInTheDocument();
  });

  test('matches the Navbar component snapshot', () => {
    const { container } = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});
