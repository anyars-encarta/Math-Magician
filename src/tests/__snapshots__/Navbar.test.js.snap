// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches the Navbar component snapshot 1`] = `
<div>
  <div
    class="navbar"
  >
    <h1
      class="header"
    >
      Encarta Math Magicians
    </h1>
    <nav>
      <li>
        <a
          class="navLink"
          href="/home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="navLink"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="navLink"
          href="/quote"
        >
          Quote
        </a>
      </li>
    </nav>
  </div>
</div>
`;
